testng-core/src/test/java/test/configuration/ConfigurationTest.java

```diff
@@ -1,98 +1,30 @@
-
-  @BeforeClass
-  public void beforeClass() {
-    assert m_beforeSuite : "beforeSuite should have run";
-    assert !m_afterSuite : "afterSuite shouldn't have run";
-    assert !m_beforeClass : "beforeClass shouldn't have run";
-    assert !m_afterClass : "afterClass shouldn't have run";
-    assert !m_beforeMethod : "beforeMethod shouldn't have run";
-    assert !m_afterMethod : "afterMethod shouldn't have run";
-    m_beforeClass = true;
-  }
-
-  @BeforeMethod
-  public void beforeMethod() {
-    assert m_beforeSuite : "beforeSuite should have run";
-    assert m_beforeClass : "beforeClass have run";
-    assert !m_afterSuite : "afterSuite shouldn't have run";
-    assert !m_afterClass : "afterClass shouldn't have run";
-    assert !m_beforeMethod : "beforeMethod shouldn't have run";
-    assert !m_afterMethod : "afterMethod shouldn't have run";
-    m_beforeMethod = true;
-  }
-
-  @AfterMethod
-  public void afterMethod() {
-    assert m_beforeSuite : "beforeSuite should have run";
-    assert m_beforeClass : "beforeClass have run";
-    assert m_beforeMethod : "beforeMethod should have run";
-    assert !m_afterSuite : "afterSuite shouldn't have run";
-    assert !m_afterClass : "afterClass shouldn't have run";
-    assert !m_afterMethod : "afterMethod shouldn't have run";
-    m_afterMethod = true;
-  }
-
-  @AfterClass
-  public void afterClass() {
-    assert m_beforeSuite : "beforeSuite should have run";
-    assert m_beforeClass : "beforeClass have run";
-    assert m_beforeMethod : "beforeMethod should have run";
-    assert m_afterMethod : "afterMethod should have run";
-    assert !m_afterClass : "afterClass shouldn't have run";
-    assert !m_afterSuite : "afterSuite shouldn't have run";
-    m_afterClass = true;
+public class ConfigurationTest extends ConfigurationBaseTest {
+  @Test
+  public void testConfiguration() {
+    testConfiguration(ConfigurationTestSample.class);
   }
 
-  @AfterSuite
-  public void afterSuite() {
-    assert m_beforeSuite : "beforeSuite should have run";
-    assert m_beforeClass : "beforeClass have run";
-    assert m_beforeMethod : "beforeMethod should have run";
-    assert m_afterMethod : "afterMethod should have run";
-    assert m_afterClass : "afterClass should have run";
-    assert !m_afterSuite : "afterSuite shouldn't have run";
-    m_afterSuite = true;
+  @Test
+  public void testMethodCallOrder() {
+    testConfiguration(MethodCallOrderTestSample.class, ExternalConfigurationClass.class);
   }
 
   @Test
-  public void verify() {
-    assert m_beforeSuite : "beforeSuite should have run";
-    assert m_beforeClass : "beforeClass have run";
-    assert m_beforeMethod : "beforeMethod should have run";
-    assert !m_afterSuite : "afterSuite shouldn't have run";
-    assert !m_afterClass : "afterClass shouldn't have run";
-    assert !m_afterMethod : "afterMethod shouldn't have run";
+  public void testSuite() {
+    testConfiguration(SuiteTestSample.class, VerifySuiteTestSample.class);
```

Comment:
In fact, `VerifySuiteTestSample` is the expected test code for `SuiteTestSample`.

Reply:
VerifySuiteTestSample.class removed